### DISCUSSION TITLE
sendsysex vs. send

### DISCUSSION THREAD

**Discussion by schoko11** [djipco/webmidi#350] [Q&A] [answered]
Hi,

i use 

```
midiOutput.send(0xf0,[ 0x00, 0x20, 0x33, 0x00, 0x7f, 0x42, 0x00, 0x32, 0x00, 0xf7]); 
```
// 0x00 0x20 0x33   is the manufacture id and "0x00" the product typ
//0xf0 0x00 0x20 0x33 0x00 0x7f [message] 0xf7  //general message layout according documentation

for sucessfully requesting sysex for the device

but i would like to simply use sendsysex (better would be realtime instead of manufacture id)
```
midiOutput.sendSysex( [0,32,51 ] ,[66, 0, 50, 0]);   //did not work
midiOutput.sendSysex( [0,32,51,0 ] ,[66, 0, 50, 0]);   //did not work
midiOutput.sendSysex( 127,[66, 0, 50, 0]);   //did not work
...
```

can someone give me a hint why this is not working ?

Thanks!

**Comment by schoko11** [comment]
Well finally found it (4 digits for the manufacturer tricked me)...
midiOutput.sendSysex( [0,32,51, 0],[127, 66, 0, 50, 0] );  


any ideas why i can't use 
midiOutput.sendSysex( 127,[127, 66, 0, 50, 0] );

or better said to be sure it is realtime (seems to be the case anyway...)

**Comment by djipco** [comment, marked as answer]
The first parameter of the [`sendSysex()`](https://webmidijs.org/api/classes/Output#sendSysex) method is the MIDI manufacturer's [identification number](https://www.midi.org/specifications-old/item/manufacturer-id-numbers). Those are either 1 or 3 bytes long. WEBMIDI.js does not verify the length of the first parameter and simply merges it with the data to be sent (second parameter).

So, if it works, it is probably because what you needed is this:

```js
midiOutput.sendSysex( [0, 32, 51], [0, 127, 66, 0, 50, 0] );
```

The `[0, 32, 51]` array identifies the manufacturer (e.g. Access Music Electronics) and the second array is the actual device-specific data to send (this varies from one manufacturer to another).

> any ideas why i can't use `midiOutput.sendSysex( 127,[127, 66, 0, 50, 0] );`

Despite their name, [Universal System Exclusive Messages](https://www.midi.org/specifications-old/item/table-4-universal-system-exclusive-messages) are NOT system exclusive. They are an extension to the MIDI standard meant to be supported by all devices. So, sending manufacturer-specific sysex messages is very different from sending "universal" sysex messages.

Hope this helps!

**Reply by schoko11** [reply]
yepp, thanks
